src/ima.js

```diff
@@ -584,9 +584,11 @@ class Ima extends BasePlugin implements IMiddlewareProvider, IAdsControllerProvi
    * @memberof Ima
    */
   _loadImaSDKLib(): Promise<*> {
+    const protocolRegex = /^(https:|http:)/i.exec(document.location.protocol);
+    const protocol = protocolRegex && protocolRegex.hasOwnProperty('index') ? protocolRegex[protocolRegex.index] : 'https:';
```

Comment:
```suggestion
    const protocol = /^(https?:)/i.test(document.location.protocol) || "https:"
```

Reply:
It's not good.
test method return true or false, so your protocol will return true instead the relevant protocol

Reply:
correct, it should have been
```suggestion
    /^(https?:)/i.test(document.location.protocol) ? document.location.protocol : "https:"
```

Reply:
and also add the `+':'` for `document.location.protocol`